payjoin-cli/src/app/mod.rs

```diff
@@ -153,13 +147,18 @@ fn http_agent_builder() -> Result<reqwest::ClientBuilder> {
     use rustls::pki_types::CertificateDer;
     use rustls::RootCertStore;
 
-    let mut local_cert_path = std::env::temp_dir();
-    local_cert_path.push(LOCAL_CERT_FILE);
-    let cert_der = std::fs::read(local_cert_path)?;
+    let cert_der = read_local_cert()?;
     let mut root_cert_store = RootCertStore::empty();
     root_cert_store.add(CertificateDer::from(cert_der.as_slice()))?;
     Ok(reqwest::ClientBuilder::new()
         .danger_accept_invalid_certs(true)
         .use_rustls_tls()
         .add_root_certificate(reqwest::tls::Certificate::from_der(cert_der.as_slice())?))
 }
+
+#[cfg(feature = "danger-local-https")]
+fn read_local_cert() -> Result<Vec<u8>> {
```

Comment:
One observation here is that compiling payjoin-cli with `v2` enabled doesn't compile anymore without also including the `danger-local-https` feature:

```
❯ RUST_LOG=debug cargo run --features=v2 -- send --fee-rate=2 $BIP21

error[E0432]: unresolved import `crate::app::read_local_cert`
   --> payjoin-cli/src/app/v2.rs:15:30
    |
15  | use crate::app::{http_agent, read_local_cert};
    |                              ^^^^^^^^^^^^^^^ no `read_local_cert` in `app`
    |
note: found an item that was configured out
   --> payjoin-cli/src/app/mod.rs:160:4
    |
160 | fn read_local_cert() -> Result<Vec<u8>> {
    |    ^^^^^^^^^^^^^^^
    = note: the item is gated behind the `danger-local-https` feature

For more information about this error, try `rustc --explain E0432`.
error: could not compile `payjoin-cli` (bin "payjoin-cli") due to 1 previous error
```

Reply:
Great catch. Thanks for testing so thoughtfully.